File: sodaslurper.cpp
#include <iostream>
using namespace std;

int main() {
	int e, f, c;
	cin >> e >> f >> c;
	
	int ans = 0;
	int count = e + f;
	while (count >= c) {
		int bottles = count / c;
		ans += bottles;
		count = (count % c) + bottles;
	}
	cout << ans;
	return 0;
}

File: sodaslurper.hs
solve :: Int -> Int -> Int -> Int
solve bottles c total =
 if bottles < c
 then total
 else solve (canBuy + remain) c newTotal
  where (canBuy, remain) = bottles `divMod` c
        newTotal = total + canBuy

main :: IO()
main = do
 line <- getLine
 let [e, f, c] = map read $ words line
 let ans = solve (e + f) c 0
 print ans
